source/tutorials/python.rst

```diff
@@ -153,13 +156,38 @@ Finally, you can run the following command to generate all the warnings:
 
 This gives you an idea about how well your code was written, you should have at least 7/10.
 
+Reduce complexity of your code (optional)
+-----------------------------------------
+
+Learning about condition statement, for loop, and various error check is commonly considered a milestone in ones journey to programming mastery.
+However, overusing these tools can make you code complex to understand for anyone, include future you.
+There's a simple way to quickly assess the complexity of your code: look on the left hand side of the code, you want the line formed by the start of the code to be as straight as possible.
+Practically speaking, you want to reduce the use of conditions and loops, and write smaller methods.
+
+.. image:: img/squint-test.jpeg
+    :width: 600px
+
+`Source <https://www.freecodecamp.org/news/how-i-helped-my-partner-learn-to-code-6e1d1953812f/>`_
+
+If you want to gain immediate, further feedback on your python code, ``sourcery`` is a plug-in that gives you immediate feedback on code complexity.
+
+Install this plugin:
+
+.. code:: bash
+
+  code --install-extension sourcery.sourcery
+
+To let sourcery give you feedback, you will need to have some part of the code in a function.
+Hover on a function you wrote and see the quality report on complexity, method length, and working memory.
+`Find the explaination of the metric here <https://github.com/sourcery-ai/sourcery/wiki/Quality-Report#code-metrics>`_.
+If the code metric shows you some sad faces, chances are you still have a lot of nested if conditions, and the function is big.
+
 Other optionnal advices
 -----------------------
 
-What we just saw are the strict minimal good practices when writing python code.
-Additionnally, you would like to add tests with `pytest <https://docs.pytest.org/en/6.2.x/>`_ (in the ``src/test`` folder).
-A good documentation on how to run your code in the ``README.md`` file is also really apreciated.
-Finally, for others to be able to reproduce your experiments, it is important to make a good `requirement file <https://pip.pypa.io/en/stable/user_guide/#requirements-files>`_ to list the software dependencies.
+A good documentation on how to run your code in the ``README.md`` file is usually apreciated.
+For others to be able to reproduce your experiments, it is important to make a good `requirement file <https://pip.pypa.io/en/stable/user_guide/#requirements-files>`_ to list the software dependencies.
+Add tests with `pytest <https://docs.pytest.org/en/6.2.x/>`_ (in the ``src/test`` folder) will help people understand the functionality and robustness of your code.
 
```

Comment:
Do we want to add here an example repository? @htwangtw you proposed to showcase: https://github.com/SIMEXP/fmriprep-denoise-benchmark
I also have some to propose, but they don't use all recommendations from this tuto haha

Reply:
The only concern I have is, it is still a WIP project. I am not sure how helpful that would be.